src/main/java/com/iridium/iridiumskyblock/listeners/EntityDamageListener.java

```diff
@@ -3,46 +3,162 @@
             Projectile projectile = (Projectile) event.getDamager();
             if (projectile.getShooter() instanceof Player) {
                 Player player = (Player) projectile.getShooter();
                 if (IridiumSkyblock.getInstance().getIslandManager().getIslandPermission(island.get(), IridiumSkyblock.getInstance().getUserManager().getUser(player), PermissionType.KILL_MOBS)) {
                     return;
                 }
-
                 event.setCancelled(true);
                 projectile.remove();
-                player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotHurtMobs.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+                if (!canGetResponse(player)) {
+                    player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotHurtMobs.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+                }
+            }
+        }
+    }
+
+    public void handleDamageBetweenPlayers(EntityDamageByEntityEvent event, Player attacker, Player victim, Island island) {
+        Configuration configuration = IridiumSkyblock.getInstance().getConfiguration();
+
+        if (!configuration.pvpSettings.pvpOnIslands) {
+            if (!canGetResponse(attacker)) {
+                attacker.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotHurtPlayer.replace("%prefix%", configuration.prefix)));
+            }
+
+            if (event.getCause().equals(EntityDamageEvent.DamageCause.PROJECTILE)) event.getDamager().remove();
+            event.setCancelled(true);
+            return;
+        }
+
+        User attackerUser = IridiumSkyblock.getInstance().getUserManager().getUser(attacker);
+        User victimUser = IridiumSkyblock.getInstance().getUserManager().getUser(victim);
+        Optional<IslandTrusted> attackerTrusted = IridiumSkyblock.getInstance().getIslandManager().getIslandTrusted(island, attackerUser);
+        Optional<IslandTrusted> victimTrusted = IridiumSkyblock.getInstance().getIslandManager().getIslandTrusted(island, victimUser);
+        boolean attackerIsMember = island.equals(attackerUser.getIsland().orElse(null)) || attackerTrusted.isPresent();
+        boolean victimIsMember = island.equals(victimUser.getIsland().orElse(null)) || victimTrusted.isPresent();
+
+        if (!configuration.pvpSettings.pvpBetweenMembers && attackerIsMember && victimIsMember) {
+            if (!canGetResponse(attacker)) {
+                attacker.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotHurtMember.replace("%prefix%", configuration.prefix)));
             }
+            if (event.getCause().equals(EntityDamageEvent.DamageCause.PROJECTILE)) event.getDamager().remove();
+            event.setCancelled(true);
         }
     }
 
+    public void handPlayerDamage(EntityDamageEvent event, Island island) {
```

Comment:
All of these methods can be private
```suggestion
    private void handlePlayerDamage(EntityDamageEvent event, Island island) {
```

Reply:
Tf how i did it to "hand" ??